src/components/TodoItem.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FaTrash } from 'react-icons/fa';
+import styles from './TodoItem.module.css';
+
+const TodoItem = (props) => {
+  const [editing, setEditing] = useState(false);
+  const {
+    id, title, completed, handleChangeProps, deleteTodoProps, setUpdate,
+  } = props;
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const viewMode = {};
+  const editMode = {};
+
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  return (
+    <li className={styles.item}>
+      <div onDoubleClick={handleEditing} style={viewMode}>
+        <input
+          type="checkbox"
+          className={styles.checkbox}
+          checked={completed}
+          onChange={() => handleChangeProps(id)}
+        />
+        <button type="button" onClick={() => deleteTodoProps(id)}>
+          <FaTrash
+            style={{
+              color: 'orangered',
+              fontSize: '16px',
+            }}
+          />
+        </button>
+        <span style={completed ? completedStyle : null}>{title}</span>
+      </div>
+      <input
+        type="text"
+        style={editMode}
+        className={styles.textInput}
+        value={title}
+        onChange={(e) => {
+          setUpdate(e.target.value, id);
+        }}
+        onKeyDown={handleUpdatedDone}
+        onBlur={() => setEditing(false)}
```

Comment:
This input form quits with the `onBlu`r event, but it doesn't automatically focus the input with the `onDoubleClick` event that "opens" the input field. This gives a somewhat confusing user experience. My suggestion is to autofocus the input form in the `onDoubleClick` event.